core/src/main/java/com/distkv/dst/core/concepts/DstSortedListsImpl.java

```diff
@@ -105,32 +68,27 @@ public void incrScore(String key, String member, int delta) {
     if (!dstKeyValueMap.containsKey(key)) {
       throw new KeyNotFoundException(key);
     }
-    LinkedList list = dstKeyValueMap.get(key);
-    if (topNum > list.size()) {
-      topNum = list.size();
+    final SortedList sortedList = dstKeyValueMap.get(key);
+    if (topNum > sortedList.getSize()) {
```

Comment:
when topNum get bigger than getSize()?

Reply:
When the user inputs the size which is bigger than getSize() probably, this is entirely possible. Our approach is to throw an exception, we can also be processed like this. I think it is all feasible.

Reply:
Good concern.

I agree that we should return all members if topNum > size.